compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -746,27 +746,30 @@ pub enum BorrowKind {
     /// cannot currently be expressed by the user and is used only in
     /// implicit closure bindings. It is needed when the closure is
     /// borrowing or mutating a mutable referent, e.g.:
-    ///
-    ///     let x: &mut isize = ...;
-    ///     let y = || *x += 5;
-    ///
+    /// ```
+    /// let mut z = 3;
+    /// let x: &mut isize = &mut z;
+    /// let y = || *x += 5;
+    /// ```
     /// If we were to try to translate this closure into a more explicit
     /// form, we'd encounter an error with the code as written:
-    ///
-    ///     struct Env { x: & &mut isize }
-    ///     let x: &mut isize = ...;
-    ///     let y = (&mut Env { &x }, fn_ptr);  // Closure is pair of env and fn
-    ///     fn fn_ptr(env: &mut Env) { **env.x += 5; }
-    ///
+    /// ```ignore HELP (this does indeed error but I'm not sure it's the right one?)
+    /// struct Env { x: & &mut isize }
```

Comment:
```suggestion
    /// ```compile_fail,E0594
    /// struct Env<'a> { x: &'a &'a mut isize }
```
and also change the initialization to `x: &x` on line 760.